lib/rubocop/cop/lint/meterpreter_commands_dependencies.rb

```diff
@@ -0,0 +1,393 @@
+module RuboCop
+  module Cop
+    module Lint
+      # `array.any?` is a simplified way to say `!array.empty?`
+      #
+      # @example
+      #   # bad
+      #   !array.empty?
+      #
+      #   # good
+      #   array.any?
+      class MeterpreterCommandDependencies < Base
+        MSG = 'Convert meterpreter api calls into meterpreter command dependencies.'.freeze
+
+        def_node_matcher :file_rm_call?, <<~PATTERN
+          (send (send (send (send nil? :client) :fs) :file) :rm)
+        PATTERN
+
+        def_node_matcher :file_ls_call?, <<~PATTERN
+          (send (send (send (send nil? :client) :fs) :file) :ls)
+        PATTERN
+
+        def_node_matcher :command_list?, <<~PATTERN
+          (lvasgn :commands
+            _)
+        PATTERN
+
+        extend AutoCorrector
+
+        def on_new_investigation
+          super
+          @current_commands = [] # TODO: Not sure if I need to keep track of current commands, as I can just add all to latest and rewrite the list
```

Comment:
You'll need to keep track of the existing commands, so you can add and offense when there's additional commands present that aren't required